server/controllers/bookmark.js

```diff
@@ -0,0 +1,26 @@
+import articleRepo from '../repository/articleRepository';
+import bookmarkRepo from '../repository/bookmarkRepository';
+import { goodHttpResponse, badHttpResponse } from '../utilities/httpResponse';
+/**
+ * Bookmark Controller class
+ */
+class Bookmark {
+  /**
+   * bookmark an article
+   * @param {object} request Request Object
+   * @param {object} response Response Object
+   * @returns {object} User Object
+   */
+  static async bookmarkArticle(request, response) {
+    const { slug } = request.params;
```

Comment:
what happens if you pass an invalid data type as the slug?

Reply:
@seyi-adeleke  we have a param validator that @wiztemple can use.

Reply:
it returns article not found.